pkg/imagestore/imagestore_test.go

```diff
@@ -175,23 +175,100 @@ var _ = Describe("HaveVersion", func() {
 		store ImageStore
 	)
 
-	BeforeEach(func() {
-		var err error
-		store, err = NewImageStore(nil, "", versions)
-		Expect(err).NotTo(HaveOccurred())
-	})
-	AfterEach(func() {
-	})
+	Describe("HaveVersion", func() {
+
+		BeforeEach(func() {
+			var err error
+			store, err = NewImageStore(nil, "", versions)
+			Expect(err).NotTo(HaveOccurred())
+		})
+		AfterEach(func() {
+		})
+
+		It("is true for versions that are present", func() {
+			Expect(store.HaveVersion("4.8", "x86_64")).To(BeTrue())
+			Expect(store.HaveVersion("4.9", "arm64")).To(BeTrue())
+		})
 
-	It("is true for versions that are present", func() {
-		Expect(store.HaveVersion("4.8", "x86_64")).To(BeTrue())
-		Expect(store.HaveVersion("4.9", "arm64")).To(BeTrue())
+		It("is false for versions that are missing", func() {
+			Expect(store.HaveVersion("4.9", "x86_64")).To(BeFalse())
+			Expect(store.HaveVersion("4.8", "arm64")).To(BeFalse())
+			Expect(store.HaveVersion("4.7", "x86_64")).To(BeFalse())
+			Expect(store.HaveVersion("4.8", "aarch64")).To(BeFalse())
+		})
 	})
 
-	It("is false for versions that are missing", func() {
-		Expect(store.HaveVersion("4.9", "x86_64")).To(BeFalse())
-		Expect(store.HaveVersion("4.8", "arm64")).To(BeFalse())
-		Expect(store.HaveVersion("4.7", "x86_64")).To(BeFalse())
-		Expect(store.HaveVersion("4.8", "aarch64")).To(BeFalse())
+	Describe("ValidateVersions", func() {
+		var (
+			ctrl       *gomock.Controller
+			mockEditor *isoeditor.MockEditor
+			dataDir    string
+		)
+
+		BeforeEach(func() {
+			var err error
+			dataDir, err = os.MkdirTemp("", "imageStoreTest")
+			Expect(err).NotTo(HaveOccurred())
+
+			ctrl = gomock.NewController(GinkgoT())
+			mockEditor = isoeditor.NewMockEditor(ctrl)
+		})
+
+		It("should error when RHCOS_IMAGES are not set i.e. versions is an empty slice", func() {
+			versions = []map[string]string{}
```

Comment:
If you're redefining versions for each test, then don't mess with the `Describe("HaveVersions"` block, just make these tests a top-level `Describe("NewImageStore"`.

Setting the value of versions here breaks the per-test scope and will mess with anything that expects versions to be defined before the test.